src/app/components/Layout/Header/Logo/index.tsx

```diff
@@ -1,15 +1,27 @@
 import React from "react";
-import { Link } from "react-router-dom";
+import { useSelector } from "react-redux";
+import { useHistory } from "react-router-dom";
+
+//selectors
+import { getAuthUser } from "../../../../redux/selectors";
 
 // styles
 import { LogoWrapper, Text } from "./components";
 
 const Logo: React.FunctionComponent = () => {
+  const history = useHistory();
+  const user = useSelector(getAuthUser);
+
+  const handlerClick = () => {
+    if (user) {
+      history.push("/");
+    } else {
+      history.push("/auth");
+    }
+  };
```

Comment:
you don't need to change use this logic here. Logo should not depend on user data. Home page redirects unauthenticated users.